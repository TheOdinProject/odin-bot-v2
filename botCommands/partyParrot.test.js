const command = require('./partyParrot');
const { randomInt } = require('./helpers');

describe('/partyparrot', () => {
  it.each([
    ['party parrot'],
    ['/party parrot'],
    ['party_parrot'],
    ['/party_parrot'],
    ['/partyparrot'],
    ['oiseau'],
    ['/oiseau'],
    ['/partyparrot'],
  ])("'%s' triggers the callback", (string) => {
    expect(command.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['part yparrot'],
    ['party'],
    ['parrot'],
    ['partiesparrot'],
    ['parrot party'],
    ['party pants parrot'],
    ['PARTY PARROT'],
    ['partying parrot'],
    ['parrot partying'],
    ['party-parrot'],
    [''],
    [' '],
    ['/ '],
  ])("'%s' does not trigger the callback", (string) => {
    expect(command.regex.test(string)).toBeFalsy();
  });

  it.each([
    ['hello /party parrot'],
    ['need me some /party parrot'],
    ['hey @odin-bot, teach me how to /party parrot'],
    ['/@odin-bot ^ /me /party_parrot /test$*'],
    ['this is a long test string with a randomly placed party parrot command'],
    ['/oiseaugrealkdmsfalnd'],
    ['someone/party_parrot'],
    ['/party parrot/xx'],
    ['/partyparrot*'],
    ['/partyparrot...'],
    ['^_/oiseau...'],
  ])("'%s' - the command can be anywhere in the string", (string) => {
    expect(command.regex.test(string)).toBeTruthy();
  });
});

// mock randomInt
jest.mock('./helpers');

describe('partyparrot snapshot', () => {
  it.each([
    ['party parrot!'],
    ['!partyparrot'],
    ['party_parrot!'],
    ['oiseau!'],
    ['!'],
  ])(
    'messages including an exclamation point always return the first index in the array',
    (string) => {
      expect(command.cb({ content: string })).toMatchSnapshot();
    },
  );

  it('when randomInt is mocked, the output remains the same', () => {
    // NOTE: if the length of the parrot array changes,
    // this for loop will need to be updated to reflect that change
    for (let i = 0; i < 8; i += 1) {
      randomInt.mockReturnValueOnce(i);

      expect(command.cb({ content: '' })).toMatchSnapshot();
    }
  });
});
