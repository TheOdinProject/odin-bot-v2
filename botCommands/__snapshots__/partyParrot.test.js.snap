// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`partyparrot snapshot messages including an exclamation point always return the first index in the array 1`] = `"https://cultofthepartyparrot.com/parrots/hd/dadparrot.gif"`;

exports[`partyparrot snapshot messages including an exclamation point always return the first index in the array 2`] = `"https://cultofthepartyparrot.com/parrots/hd/dadparrot.gif"`;

exports[`partyparrot snapshot messages including an exclamation point always return the first index in the array 3`] = `"https://cultofthepartyparrot.com/parrots/hd/dadparrot.gif"`;

exports[`partyparrot snapshot messages including an exclamation point always return the first index in the array 4`] = `"https://cultofthepartyparrot.com/parrots/hd/dadparrot.gif"`;

exports[`partyparrot snapshot messages including an exclamation point always return the first index in the array 5`] = `"https://cultofthepartyparrot.com/parrots/hd/dadparrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 1`] = `"https://cultofthepartyparrot.com/parrots/hd/dadparrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 2`] = `"http://cultofthepartyparrot.com/parrots/parrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 3`] = `"http://cultofthepartyparrot.com/parrots/fiestaparrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 4`] = `"http://cultofthepartyparrot.com/parrots/explodyparrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 5`] = `"http://cultofthepartyparrot.com/parrots/slomoparrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 6`] = `"http://cultofthepartyparrot.com/parrots/hd/dealwithitparrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 7`] = `"http://cultofthepartyparrot.com/parrots/tripletsparrot.gif"`;

exports[`partyparrot snapshot when randomInt is mocked, the output remains the same 8`] = `"http://emojis.slackmojis.com/emojis/images/1450738632/246/leftshark.png"`;
